src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -398,7 +398,8 @@ private static bool ContainsHttpMethod(List<string> httpMethods, string httpMeth
         {
             for (var i = 0; i < httpMethods.Count; i++)
             {
-                if (string.Equals(httpMethods[i], httpMethod, StringComparison.OrdinalIgnoreCase))
+                var item = httpMethods[i];
+                if (object.ReferenceEquals(item, httpMethod) || StringComparer.OrdinalIgnoreCase.Equals(item, httpMethod))
```

Comment:
Do two loops? First with `.ReferenceEquals`, then second with `OrdinalIgnoreCase.Equal`; then you do all the fast ptr compare checks before falling back to the slower `OrdinalIgnoreCase` checks

Reply:
If you want a bit wild you can use `System.Runtime.InteropServices.CollectionsMarshal.AsSpan(httpMethods)` to get a span of the `List<T>` elements to loop through which is faster than looping through the List as the Span can skip the bounds checks for a regular `for` loop.

```csharp
var methods = CollectionsMarshal.AsSpan(httpMethods);
for (var i = 0; i < methods .Length; i++)
{
    var item = methods[i];
    if (object.ReferenceEquals(item, httpMethod))
    {
        return true;
    }
}

for (var i = 0; i < methods .Length; i++)
{
    var item = methods[i];
    if (StringComparer.OrdinalIgnoreCase.Equals(item, httpMethod))
    {
        return true;
    }
}

return false;
```

Reply:
or we could store `httpMethods` as an array and skip the AsSpan

Reply:
Works too